### PR TITLE
sound: Refactor OpenAL backend

### PR DESCRIPTION
**Pull request by Cacodemon345**
Summary
=======
sound: Refactor OpenAL backend

Also adds the ability to create extra sound sources for both OpenAL and XAudio2.

Checklist
=========
* [X] I have discussed this with core contributors already
* [ ] This pull request requires changes to the ROM set
  * [ ] I have opened a roms pull request - https://github.com/86Box/roms/pull/changeme/

References
==========
None.

### CODE REVIEW

File: src/include/86box/sound.h [hunk 1]
```diff
@@ -71,6 +77,8 @@ extern void closeal(void);
 extern void inital(void);
 extern void givealbuffer(void *buf);
 extern void givealbuffer_cd(void *buf);
+extern void givealbuffer_source(void *buf, al_source* src, int size, int freq);
+extern al_source* al_create_sound_source(uint32_t freq, uint32_t buffer_size_mono);
```
Comment (dhrdlicka) [inline src/include/86box/sound.h:81]: Given that you are already messing with these parts, would it be possible to change the function names to something a bit more generic?

**Review by dhrdlicka — CHANGES_REQUESTED** [review]

File: src/include/86box/sound.h (hunk 1 above)
Comment (Cacodemon345) [reply src/include/86box/sound.h:81]: al makes more sense since it's the audio library we are interfacing with.

Comment (dhrdlicka) [reply src/include/86box/sound.h:81]: not if XAudio2/FAudio is enabled...

Comment (Cacodemon345) [reply src/include/86box/sound.h:81]: They count as audio libraries as well.

Comment (lemondrops) [reply src/include/86box/sound.h:81]: Except that `al` in these functions' names refers to Open**AL** specifically, not just any audio library.

Comment (OBattler) [reply src/include/86box/sound.h:81]: Well, it turns out this PR has more problems: My Windows 98 SE VM drops from 100% to 92% on the POST screen when the NCR 53c8xx SCSI BIOS runs, with this PR, so my intuition that this was going to slightly slowdown the emulator overall, has been correct.

**Comment by Cacodemon345** [comment]
I suspect this will cause performance issues down the line when the extra sources feature start to be used. Closing.

**Pull request closed without merging**